src/libecalc/application/energy_calculator.py

```diff
@@ -230,34 +220,14 @@ def evaluate_emissions(self) -> dict[str, dict[str, EmissionResult]]:
         """
         emission_results: dict[str, dict[str, EmissionResult]] = {}
         for energy_component in self._energy_model.get_energy_components():
-            if isinstance(energy_component, FuelConsumerDTO | GeneratorSetDTO):
-                fuel_model = FuelModel(energy_component.fuel)
-                fuel_usage = self._get_context(energy_component.id).get_fuel_usage()
-                emission_results[energy_component.id] = fuel_model.evaluate_emissions(
-                    expression_evaluator=self._expression_evaluator,
-                    fuel_rate=np.asarray(fuel_usage.values),
-                )
-            elif isinstance(energy_component, ConsumerSystemDTO):
-                if energy_component.consumes == ConsumptionType.FUEL:
-                    fuel_model = FuelModel(energy_component.fuel)
-                    fuel_usage = self._get_context(energy_component.id).get_fuel_usage()
-                    emission_results[energy_component.id] = fuel_model.evaluate_emissions(
-                        expression_evaluator=self._expression_evaluator,
-                        fuel_rate=np.asarray(fuel_usage.values),
-                    )
-            elif isinstance(energy_component, YamlDirectTypeEmitter | YamlOilTypeEmitter):
-                venting_emitter_results = {}
-                emission_rates = energy_component.get_emissions(
+            if isinstance(energy_component, Emitter):
+                emission_result = energy_component.evaluate_emissions(
+                    energy_context=self._get_context(energy_component.id),
+                    energy_model=self._energy_model,
                     expression_evaluator=self._expression_evaluator,
-                    regularity=self._energy_model.get_regularity(energy_component.id),
                 )
 
-                for emission_name, emission_rate in emission_rates.items():
-                    emission_result = EmissionResult(
-                        name=emission_name,
-                        periods=self._expression_evaluator.get_periods(),
-                        rate=emission_rate,
-                    )
-                    venting_emitter_results[emission_name] = emission_result
-                emission_results[energy_component.id] = venting_emitter_results
+                if emission_result is not None:
```

Comment:
why can it be None?

Reply:
Because of consumer system v2 which can be both fuel and el, so a class that might be an emitter.